### PR TITLE
Add PartnerDatabaseName to New-AzSqlDatabaseSecondary cmdlet

### PR DESCRIPTION
**Pull request by bradrich-msft**
## Description

This adds a new parameter to an existing cmdlet.  The API/SDK has long-supported this parameter and it was just not yet added to Powershell.

## Checklist

- [x ] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [x] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by adxsdkps** [comment]
Can one of the admins verify this patch?

File: src/Sql/Sql/Replication/Cmdlet/NewAzureSqlDatabaseSecondary.cs [hunk 1]
```diff
@@ -187,6 +195,7 @@ protected override IEnumerable<AzureReplicationLinkModel> ApplyUserInputToModel(
                 DatabaseName = this.DatabaseName,
                 PartnerResourceGroupName = this.PartnerResourceGroupName,
                 PartnerServerName = this.PartnerServerName,
+                PartnerDatabaseName = string.IsNullOrWhiteSpace(this.PartnerDatabaseName) ? this.DatabaseName : this.PartnerDatabaseName,
```
Comment (isra-fel) [inline src/Sql/Sql/Replication/Cmdlet/NewAzureSqlDatabaseSecondary.cs]: What is the meaning of setting `PartnerDatabaseName` the same as `DataBaseName`?

Comment (bradrich-msft) [reply src/Sql/Sql/Replication/Cmdlet/NewAzureSqlDatabaseSecondary.cs]: Hi.  This is current behavior.  For backwards compatibility, if client does not specify PartnerDatabaseName, they will get the same behavior as today which always uses DatabaseName for the partner.  A name is required no matter what.

**Comment by bradrich-msft** [comment]
Hi @isra-fel,

I have answered your inline comment.  Regarding usage examples, I have added example usage of the new parameter in the test file being updated with this pull request.  Was there some other location to put examples?

**Comment by bradrich-msft** [comment]
Hi @isra-fel, 

I have added a usage example to the help file.

**Pull request merged**